lib/livebook_web/live/app_live.ex

```diff
@@ -6,37 +6,41 @@ defmodule LivebookWeb.AppLive do
   alias Livebook.Notebook.Cell
 
   @impl true
-  def mount(%{"slug" => slug}, _session, socket) do
-    case Livebook.Apps.fetch_session_by_slug(slug) do
-      {:ok, %{pid: session_pid, id: session_id}} ->
+  def mount(%{"slug" => slug}, _session, socket) when socket.assigns.app_authenticated? do
+    {:ok, %{pid: session_pid, id: session_id}} = Livebook.Apps.fetch_session_by_slug(slug)
```

Comment:
```suggestion
    %{pid: session_pid, id: session_id} = socket.assigns.app_settings
```

?

Reply:
Oh wait, we fetch the session info here, not the settings 🙈

Reply:
ah, i got confused :D